stm/stm.ml

```diff
@@ -1950,10 +1949,15 @@ end = struct (* {{{ *)
         | Evd.Evar_empty -> RespNoProgress
         | Evd.Evar_defined t ->
             let t = Evarutil.nf_evar sigma t in
-            if Evarutil.is_ground_term sigma t then
+            let evars = Evarutil.undefined_evars_of_term sigma t in
+            if Evar.Set.is_empty evars then
               let t = EConstr.Unsafe.to_constr t in
               RespBuiltSubProof (t, Evd.evar_universe_context sigma)
-            else CErrors.user_err ~hdr:"STM" Pp.(str"The solution is not ground")
+            else
+              CErrors.user_err ~hdr:"STM"
+                Pp.(str"The par: selector requires a tactic that makes no progress or fully" ++
+                    str" solves all goals and leaves no unresolved existentials. The following" ++
```

Comment:
I wouldn't say that the tactic solves all goals, it only solves the 1 goal it is given and leaves the other to the parallel runs.

Reply:
```suggestion
                    str" solves the goal and leaves no unresolved existential variables. The following" ++
```